File: 83-remove-duplicates-from-sorted-list/remove-duplicates-from-sorted-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:
        # Initialize a pointer to the head of the list
        current = head
        
        # Traverse the list
        while current is not None and current.next is not None:
            if current.val == current.next.val:
                # Skip the next node by updating the 'next' pointer
                current.next = current.next.next
            else:
                # Move to the next node
                current = current.next
        
        return head

File: 83-remove-duplicates-from-sorted-list/remove-duplicates-from-sorted-list.rb
def delete_duplicates(head)
  current_node = head
  before_node = nil
  head_of_sequence = nil
  while current_node
    # Check if the value of the current node is equal to the value of the previous node
    if before_node && before_node.val == current_node.val
      # If so, link the previous node to the next node, effectively removing the current node
      before_node.next = current_node.next
    else
      # If not, update the before_node to the current_node
      before_node = current_node
    end
    # Move to the next node in the list
    current_node = current_node.next
  end
  # Return the head of the list after removing duplicates
  head
end
